model/task/generate_tasks_estimations.go

```diff
@@ -0,0 +1,72 @@
+package task
+
+import (
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/utility"
+	adb "github.com/mongodb/anser/db"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+const (
+	numRevisionsToSearch = 10
```

Comment:
Is it a concern at all that the query is by # revisions and not by time recency? I believe the scheduler's task estimation is based on tasks that have recently run, so if a project doesn't receive frequent commits or has a setup like lots of periodic builds (but not a lot of commit activity), it might have an estimate that's pretty far off for patches.

If we're okay with the estimate potentially being off, I'm alright with doing it like this.

Reply:
That's fair, IMO in that case we should be fine to instead copy the [estimated duration query](https://gist.github.com/hadjri/b31d0dc3bde368cdcd204307c0c63def#file-expected_duration-go-L30-L44). @bynn I know we had discussions on the query being slow but I believe this should be fine since this estimation query is already running for all tasks and hasn't been an issue.

Reply:
updated to use something more similar to estimated duration